### PR TITLE
DIP-164 Added support for plurals

### PR DESCRIPTION
**Pull request by cameron-white-janrain**
Tell me what you think about the coding style. Its a bit functional...

### CODE REVIEW

File: bluekai/bluekai_tsv.py [hunk 1]
```diff
@@ -1,38 +1,63 @@
...
-    items = filter(lambda pair: pair[0] != "uuid", items)
-    items = map(lambda pair: "{}={}".format(getKey(pair[0], keys), getValue(pair[1])), items)
-    items = str.join('|', items)
-
-    row = "{}\t{}\n".format(uuid, items)
+            return path
+
+    # Compile (path, regex) for each path
+    regex_mappings = [
+        (path, re.compile(path.replace('.', r'(?:\.|\.\d+\.)')))
```
Comment (ebeesonjanrain) [inline bluekai/bluekai_tsv.py]: would like a comment explaining why the need for the regex

Comment (cameron-white-janrain) [inline bluekai/bluekai_tsv.py]: OK

File: bluekai/bluekai_tsv.py [hunk 2]
```diff
@@ -1,38 +1,63 @@
...
+        for path in (paths or [])
+    ]
+
+    try:
+
+        walked_record = walk(record)
+
+        # (path, walked_path, value) of each walked_record that matches a
+        # regex and add path from regex_mapping.
+        walked_record = (
```
Comment (ebeesonjanrain) [inline bluekai/bluekai_tsv.py]: perhaps split this into separate generators instead of this compound one

File: requirements.txt [hunk 3]
```diff
@@ -7,3 +8,4 @@ janrain-datalib==0.1.3
 pynamodb==1.4.4
 paramiko==1.15.2
 freezegun==0.3.6
+pse-tools==1.0.1.dev0
```
Comment (ebeesonjanrain) [inline requirements.txt]: is the walk function in pse-tools not released yet?

**Comment by ebeesonjanrain** [comment]
Coding style looks good to me.

**Comment by BCox-Janrain** [comment]
Looks good. :+1:

**Pull request merged**